.github/workflows/main.yml

```diff
@@ -99,7 +99,14 @@ jobs:
           npm config set //registry.npmjs.org/:_authToken $NPM_AUTH_TOKEN
           # print the NPM user name for validation
           npm whoami
-          npm publish --otp="$(oathtool -b --totp $NPM_OTP)"
+          VERSION=$(npm -s run env echo '$npm_package_version')
+          # Only publish stable versions to the latest tag
+          if [[ "$VERSION" =~ ^[^-]+$ ]]; then
+            NPM_TAG="latest"
+          else
+            NPM_TAG="beta"
+          fi
+          npm publish --otp="$(oathtool -b --totp $NPM_OTP)" --tag $NPM_TAG
```

Comment:
Nit: Let's print the tag we're publishing to so we can debug from CI logs if needed?